src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
 export interface Figure {
-
+  color: string,
+  getArea(): number,
+  a?: number,
+  b?: number,
+  c?: number,
+  radius?: number,
+  width?: number,
+  height?:number,
+  shape: string,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: string = 'triangle';
 
-}
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.getArea();
+  }
 
-export class Circle {
+  getArea():number {
+    if (this.a + this.b > this.c) {
```

Comment:
Move the triangle validation logic to the constructor. Throw an error if the triangle is invalid, and remove the checks in the getArea method.